### PR TITLE
First cut at v1beta3 FAQ and versioning rationale.

### LINKED ISSUE #3670: Announce that v1beta1 and v1beta2 are going away when v1beta3 is exposed by default
it's impossible to GET replication controllers on v3. on v1 on the same server it works fine.

example request: GET on http://localhost:8080/api/v1beta3/replicationControllers returns 404

http://localhost:8080/api/v1beta1/replicationControllers returns 200

### PR DESCRIPTION
**Pull request by bgrant0607**
Fixes #3670, #6272, #2695.

@lavalamp @thockin @smarterclayton @nikhiljindal

### CODE REVIEW

**Comment by bgrant0607** [comment]
Once this is merged, I'll send an announcement to kubernetes-announce@googlegroups.com announcing v1beta3 and EOL for v1beta1/2.

We can grow the "conversion tips" section as people think of more gotchas.

File: docs/api.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Some important differences between v1beta1/2 and v1beta3:
+* The resource `id` is now called `name`.
+* `name`, `labels`, `annotations`, and other metadata are now nested in a map called `metadata`
+* `desiredState` is now called `spec`, and `currentState` is now called `status`
+* `/minions` has been moved to `/nodes`, and the resource has kind `Node`
+* The namespace is required (for all namespaced resources) and has moved from a URL parameter to the path: `/api/v1beta3/namespaces/{namespace}/{resource_collection}/{resource_name}`
+* The container `entrypoint` has been renamed to `command`, and `command` has been renamed to `args`.
+* Container, volume, and node resources are expressed as nested maps (e.g., `resources{cpu:1}`) rather than as individual fields, and resource values support [scaling suffixes](resources.md#resource-quantities) rather than fixed scales (e.g., milli-cores).
+* Restart policy is represented simply as a string (e.g., "Always") rather than as a nested map ("always{}").
+* The volume `source` is inlined into `volume` rather than nested.
```
Comment (smarterclayton) [inline docs/api.md]: To watch for changes to a resource, open an HTTP or Websocket connection to the collection URL and provide the `?watch=true` query parameter along with the desired resourceVersion query parameter to watch from.

Comment (smarterclayton) [inline docs/api.md]: The names of all resource collections are now lower cased - instead of "replicationControllers", provide "replicationcontrollers".

Comment (bgrant0607) [inline docs/api.md]: Good points. Stole your text.

**Comment by lavalamp** [comment]
LGTM - it's easy to make a followup if anyone can think of more to say.

File: docs/api.md (hunk 1 above)
Comment (nikhiljindal) [inline docs/api.md]: Another change:
"labels" query parameter has been changed to "label-selector"

**Comment by thockin** [comment]
I am OOO tomorrow, can you add the hostDir -> hostPath change here, too?

**Pull request merged**